marklogic-data-hub-spark-connector/src/test/java/com/marklogic/hub/spark/sql/sources/v2/writer/HandleAbortedWriterTest.java

```diff
@@ -31,8 +32,10 @@ void dataSourceWriterIsAborted() {
         assertEquals(2, getFruitCount(), "Just verifying the two fruits were written");
 
         dataSourceWriter.abort(null);
-        assertEquals("canceled", getJobDocumentStatus(), "If the DataSourceWriter is aborted for any reason, the job " +
+        assertTrue(getJobDocumentStatus().equals("canceled") ||
```

Comment:
Is the "or" check here only because it will be "started" on ML 9? If so, this should first do a check of the ML version to see if it's ML 9, and use that to determine what the expected status should be. Same goes for the other "or" checks below.

Reply:
1. The failures are not consistent in 9.0-x. In WriteJobWithCustomEndpointsTest , whenever custom endpoint is used to initialize write , we are able to update the job document (regardless of whether custom endpoint is used to finalize write or not). 
2. If default endpoints are used, we can create job doc but are not able to update it.